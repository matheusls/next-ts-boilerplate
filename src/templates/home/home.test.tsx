import { render, screen } from '@testing-library/react';

import Home from './home';

const titleRegex = /next.js, typescript and styled-components boilerplate/i;
const title = titleRegex.source;

describe('<Home /> component', () => {
  it('should match snapshot', () => {
    const { container } = render(<Home title={title} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render component correctly', () => {
    render(<Home title={title} />);

    expect(
      screen.getByRole('heading', {
        name: titleRegex,
      }),
    ).toBeInTheDocument();

    expect(
      screen.getByText(
        /a simple project starter to work with typescript, react, next.js and styled-components./i,
      ),
    ).toBeInTheDocument();
  });
});
