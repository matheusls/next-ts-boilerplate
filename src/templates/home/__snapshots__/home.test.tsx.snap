// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> component should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #f0f0f0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 2rem 1rem;
  text-align: center;
  width: 100%;
}

<main
  class="c0"
>
  <h1>
    next.js, typescript and styled-components boilerplate
  </h1>
  <p>
    A simple project starter to work with TypeScript, React, Next.js and Styled-Components.
  </p>
</main>
`;
